interfaces/kmod/spec.go

```diff
@@ -0,0 +1,93 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package kmod
+
+import (
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+// Specification assists in collecting kernel modules associated with an interface.
+//
+// Unlike the Backend itself (which is stateless and non-persistent) this type
+// holds internal state that is used by the kmod backend during the interface
+// setup process.
+type Specification struct {
+	Modules map[string]bool
+}
+
+// AddModule adds a kernel module, trimming spaces and ignoring duplicated modules.
+func (spec *Specification) AddModule(module string) error {
+	m := strings.TrimSpace(module)
+	if len(m) > 0 {
```

Comment:
You could just write `if len(m) == 0 { return nil }` here (or given that its a string even better: `if m == "" { return nil }`. This is more common in go (return early and have the rest of the function in the normal function scope instead of on a if).